api/mapping/views.py

```diff
@@ -545,35 +646,171 @@ def password_reset_request(request):
     password_reset_form = PasswordResetForm()
-    return render(request=request,
-                  template_name="/registration/password_reset.html",
-                  context={"password_reset_form": password_reset_form})
+    return render(
+        request=request,
+        template_name="/registration/password_reset.html",
+        context={"password_reset_form": password_reset_form},
+    )
 
 
 def load_omop_fields(request):
-    omop_table_id = request.GET.get('omop_table')
-    omop_fields = OmopField.objects.filter(table_id=omop_table_id).order_by('field')
-    return render(request, 'mapping/omop_table_dropdown_list_options.html', {'omop_fields': omop_fields})
+    omop_table_id = request.GET.get("omop_table")
+    omop_fields = OmopField.objects.filter(table_id=omop_table_id).order_by("field")
+    return render(
+        request,
+        "mapping/omop_table_dropdown_list_options.html",
+        {"omop_fields": omop_fields},
+    )
+
+
+def testusagi(request, scan_report_id):
+
+    results = run_usagi(scan_report_id)
+    print(results)
+    context = {}
+
+    return render(request, "mapping/index.html", context)
+
+
+def merge_dictionary(request):
+
+    # Grab the scan report ID
+    search_term = request.GET.get("search", None)
+    print('SEARCH TERM >>> ', search_term)
+
+    # Grab the appropriate data dictionary which is built when a scan report is uploaded
+    dictionary = DataDictionary.objects.filter(source_value__scan_report_field__scan_report_table__scan_report__id=search_term).filter(source_value__scan_report_field__is_patient_id=False).filter(source_value__scan_report_field__is_date_event=False).filter(source_value__scan_report_field__is_ignore=False).exclude(source_value__value='List truncated...')
+    
+    # Convert QuerySet to dataframe
+    dict_df = pd.DataFrame.from_dict(dictionary.values(
+                                            "source_value__scan_report_field__scan_report_table__scan_report__data_partner__name",
+                                            "source_value__scan_report_field__scan_report_table__scan_report__dataset",
+                                            "source_value__scan_report_field__scan_report_table__name",
+                                            "source_value__scan_report_field__name",
+                                            "source_value__value",
+                                            "source_value__frequency",
+                                            "dictionary_field_description",
+                                            "dictionary_value_description",
+                                            )
+                                        )
+
+    # Name columns
+    dict_df.columns = [
+        "DataPartner",
+        "DataSet",
+        "Table",
+        "Field",
+        "Value",
+        "Frequency",
+        "FieldDesc",
+        "ValueDescription",
+    ]
+
+    dict_df.to_csv('/data/TEMP_internal_dictionary.csv')
```

Comment:
as before, double check can't get screwed up when two instances running? Add some sort of hash/id to the name?

Reply:
Ah balls, forgot to take these out. The `to_csv()` calls aren't required, they were just there for debugging.